src/masternodeman.cpp

```diff
@@ -740,6 +740,48 @@ std::pair<CService, std::set<uint256> > CMasternodeMan::PopScheduledMnbRequestCo
     return std::make_pair(pairFront.first, setResult);
 }
 
+void CMasternodeMan::ProcessQueuedMnbRequests(CConnman& connman)
+{
+    std::pair<CService, std::set<uint256> > p = mnodeman.PopScheduledMnbRequestConnection();
+    if (!(p.first == CService() || p.second.empty())) {
+        if (connman.IsMasternodeOrDisconnectRequested(p.first)) return;
+        mnbQueue.push_back(std::make_pair(GetTime(), p));
+        connman.AddPendingMasternode(p.first);
+    }
+
+    if (!mnbQueue.empty()) {
+        // wait at least 5 seconds since we tried to open corresponding connection
```

Comment:
Hmm sounds like it go wrong or at least take more time then actually needed in some cases.

What about the other way around? You always loop through all fully connected nodes and for each node that is in mnbQueue you send the messages and remove the entry from mnbQueue.
After that loop, you go through mnbQueue and remove all entries that didn't get processed for some time already (which means connection failed).
This way the good nodes are processed ASAP, while the bad nodes fall out of the queue (which is not really a queue anymore) after some time.

Same applies to ProcessQueuedMnvRequests and ProcessQueuedDsa